File: app/src/main/java/net/nonylene/photolinkviewer/fragment/WifiFragment.java
package net.nonylene.photolinkviewer.fragment;

import android.content.Intent;
import android.os.Bundle;
import android.preference.ListPreference;
import android.preference.Preference;

import net.nonylene.photolinkviewer.R;
import net.nonylene.photolinkviewer.dialog.BatchDialogFragment;

public class WifiFragment extends PreferenceSummaryFragment {

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        addPreferencesFromResource(R.xml.quality_setting_wifi);
        // on click batch
        Preference preference = findPreference("quality_wifi_batch");
        preference.setOnPreferenceClickListener(new Preference.OnPreferenceClickListener() {
            @Override
            public boolean onPreferenceClick(Preference preference) {
                // batch dialog
                BatchDialogFragment batchDialogFragment = new BatchDialogFragment();
                batchDialogFragment.setTargetFragment(WifiFragment.this, 1);
                batchDialogFragment.show(getFragmentManager(), "batch");
                return false;
            }
        });
    }

    @Override
    public void onActivityResult(int requestCode, int resultCode, Intent data) {
        // batch dialog listener
        super.onActivityResult(requestCode, resultCode, data);
        switch (requestCode) {
            case 1:
                batchSelected(resultCode);
                break;
        }
    }

    private void batchSelected(int resultCode) {
        // change preferences in a lump
        ListPreference flickrPreference = (ListPreference) findPreference("flickr_quality_wifi");
        ListPreference twitterPreference = (ListPreference) findPreference("twitter_quality_wifi");
        ListPreference twipplePreference = (ListPreference) findPreference("twipple_quality_wifi");
        ListPreference imglyPreference = (ListPreference) findPreference("imgly_quality_wifi");
        ListPreference instagramPreference = (ListPreference) findPreference("instagram_quality_wifi");
        ListPreference nicoPreference = (ListPreference) findPreference("nicoseiga_quality_wifi");
        switch (resultCode) {
            case 0:
                flickrPreference.setValue("original");
                twitterPreference.setValue("original");
                twipplePreference.setValue("original");
                imglyPreference.setValue("full");
                instagramPreference.setValue("large");
                nicoPreference.setValue("original");
                break;
            case 1:
                flickrPreference.setValue("large");
                twitterPreference.setValue("large");
                twipplePreference.setValue("large");
                imglyPreference.setValue("large");
                instagramPreference.setValue("large");
                nicoPreference.setValue("large");
                break;
            case 2:
                flickrPreference.setValue("medium");
                twitterPreference.setValue("medium");
                twipplePreference.setValue("thumb");
                imglyPreference.setValue("medium");
                instagramPreference.setValue("medium");
                nicoPreference.setValue("medium");
                break;
        }


    }
}

File: app/src/main/java/net/nonylene/photolinkviewer/fragment/WifiFragment.kt
package net.nonylene.photolinkviewer.fragment

import android.content.Intent
import android.os.Bundle
import android.preference.ListPreference

import net.nonylene.photolinkviewer.R
import net.nonylene.photolinkviewer.dialog.BatchDialogFragment

class WifiFragment : PreferenceSummaryFragment() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        addPreferencesFromResource(R.xml.quality_setting_wifi)
        // on click batch
        findPreference("quality_wifi_batch").setOnPreferenceClickListener {
            // batch dialog
            BatchDialogFragment().apply {
                setTargetFragment(this@WifiFragment, 1)
                show(this@WifiFragment.fragmentManager, "batch")
            }
            false
        }
    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        // batch dialog listener
        super.onActivityResult(requestCode, resultCode, data)
        when (requestCode) {
            1 -> batchSelected(resultCode)
        }
    }

    private fun batchSelected(resultCode: Int) {
        // change preferences in a lump
        val flickrPreference = findPreference("flickr_quality_wifi") as ListPreference
        val twitterPreference = findPreference("twitter_quality_wifi") as ListPreference
        val twipplePreference = findPreference("twipple_quality_wifi") as ListPreference
        val imglyPreference = findPreference("imgly_quality_wifi") as ListPreference
        val instagramPreference = findPreference("instagram_quality_wifi") as ListPreference
        val nicoPreference = findPreference("nicoseiga_quality_wifi") as ListPreference
        val tumblrPreference = findPreference("tumblr_quality_wifi") as ListPreference

        when (resultCode) {
            0 -> {
                flickrPreference.value = "original"
                twitterPreference.value = "original"
                twipplePreference.value = "original"
                imglyPreference.value = "full"
                instagramPreference.value = "large"
                nicoPreference.value = "original"
                tumblrPreference.value = "original"
            }
            1 -> {
                flickrPreference.value = "large"
                twitterPreference.value = "large"
                twipplePreference.value = "large"
                imglyPreference.value = "large"
                instagramPreference.value = "large"
                nicoPreference.value = "large"
                tumblrPreference.value = "large"
            }
            2 -> {
                flickrPreference.value = "medium"
                twitterPreference.value = "medium"
                twipplePreference.value = "large"
                imglyPreference.value = "medium"
                instagramPreference.value = "medium"
                nicoPreference.value = "medium"
                tumblrPreference.value = "medium"
            }
        }
    }
}
